test/parallel/test-worker-heap-snapshot.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+// Flags: --expose-internals
+const common = require('../common');
+const assert = require('assert');
+const { Worker } = require('worker_threads');
+const { HeapSnapshotStream } = require('internal/heap_utils');
+const { kHandle } = require('internal/worker');
+
+const worker = new Worker(__filename);
+const snapShotResult = { ondone: () => {} };
+worker[kHandle].takeHeapSnapshot = common.mustCall(() => snapShotResult);
+
+worker.on('online', common.mustCall(() => {
+  const snapShotResponse = worker.getHeapSnapshot();
+  snapShotResult.ondone({});
+
+  snapShotResponse.then(common.mustCall((heapSnapshotStream) => {
+    assert.ok(heapSnapshotStream instanceof HeapSnapshotStream);
```

Comment:
Can we maybe read the stream here and make some basic assertions about the contents, rather than asserting that it is of a specific internal type? Even just making sure that it's valid JSON would be much more valuable in making sure that this feature actually works as intended from a user perspective

Reply:
Peeeerfect! I’m gonna change it

Reply:
done!